pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go

```diff
@@ -0,0 +1,527 @@
+				klog.V(4).Infof("Unable to look up PVC info for %s/%s, assuming PVC doesn't match predicate when counting limits: %v", namespace, pvcName, err)
+				continue
+			}
+
+			pvName := pvc.Spec.VolumeName
+			if pvName == "" {
+				// PVC is not bound. It was either deleted and created again or
+				// it was forcefully unbound by admin. The pod can still use the
+				// original PV where it was bound to, so we count the volume if
+				// it belongs to the running predicate.
+				if pl.matchProvisioner(pvc) {
+					klog.V(4).Infof("PVC %s/%s is not bound, assuming PVC matches predicate when counting limits", namespace, pvcName)
+					filteredVolumes[pvID] = true
+				}
+				continue
+			}
+
+			pv, err := pl.pvLister.Get(pvName)
+			if err != nil || pv == nil {
+				// If the PV is invalid and PVC belongs to the running predicate,
+				// log the error and count the PV towards the PV limit.
+				if pl.matchProvisioner(pvc) {
+					klog.V(4).Infof("Unable to look up PV info for %s/%s/%s, assuming PV matches predicate when counting limits: %v", namespace, pvcName, pvName, err)
+					filteredVolumes[pvID] = true
+				}
+				continue
+			}
+
+			if id, ok := pl.filter.FilterPersistentVolume(pv); ok {
+				filteredVolumes[id] = true
+			}
+		}
+	}
+
+	return nil
+}
+
+// matchProvisioner helps identify if the given PVC belongs to the running predicate.
+func (pl *NonCSILimits) matchProvisioner(pvc *v1.PersistentVolumeClaim) bool {
+	if pvc.Spec.StorageClassName == nil {
+		return false
+	}
+
+	storageClass, err := pl.scLister.Get(*pvc.Spec.StorageClassName)
+	if err != nil || storageClass == nil {
+		return false
+	}
+
+	return pl.filter.MatchProvisioner(storageClass)
+}
+
+// getMaxVolLimitFromEnv checks the max PD volumes environment variable, otherwise returning a default value.
+func getMaxVolLimitFromEnv() int {
+	if rawMaxVols := os.Getenv(predicates.KubeMaxPDVols); rawMaxVols != "" {
+		if parsedMaxVols, err := strconv.Atoi(rawMaxVols); err != nil {
+			klog.Errorf("Unable to parse maximum PD volumes value, using default: %v", err)
+		} else if parsedMaxVols <= 0 {
+			klog.Errorf("Maximum PD volumes must be a positive value, using default")
+		} else {
+			return parsedMaxVols
+		}
+	}
+
+	return -1
+}
+
+// VolumeFilter contains information on how to filter PD Volumes when checking PD Volume caps.
+type VolumeFilter struct {
+	// Filter normal volumes
+	FilterVolume           func(vol *v1.Volume) (id string, relevant bool)
+	FilterPersistentVolume func(pv *v1.PersistentVolume) (id string, relevant bool)
+	// MatchProvisioner evaluates if the StorageClass provisioner matches the running predicate
+	MatchProvisioner func(sc *storage.StorageClass) (relevant bool)
+	// IsMigrated returns a boolean specifying whether the plugin is migrated to a CSI driver
+	IsMigrated func(csiNode *storage.CSINode) bool
+}
+
+// EBSVolumeFilter is a VolumeFilter for filtering AWS ElasticBlockStore Volumes.
+var EBSVolumeFilter = VolumeFilter{
```

Comment:
any reason why this is a global variable? can't we have it defined inside ```NewEBS()``` function?